app/src/main/java/com/verygoodsecurity/demoapp/date_range_activity/DateRangeActivity.kt

```diff
@@ -0,0 +1,183 @@
+    @SuppressLint("SetTextI18n")
+    override fun onResponse(response: VGSResponse?) {
+        Log.d("Test", "CODE: ${response?.code.toString()}")
+        binding.tvResponseCode.text = "CODE: ${response?.code.toString()}"
+        setLoading(false)
+        this.response = when (response) {
+            is VGSResponse.SuccessResponse -> readShortResponse(response.body)
+            is VGSResponse.ErrorResponse -> response.body ?: response.localizeMessage
+            else -> throw IllegalArgumentException("Not implemented.")
+        }
+        updateCodeExample()
+    }
+
+    private fun readShortResponse(body: String?): String {
+        return try {
+            JSONObject(body ?: "").getJSONObject("json").toString(4)
+        } catch (e: Exception) {
+            ""
+        }
+    }
+
+    // Use this callback to track input fields state change
+    override fun onStateChange(state: FieldState) {
+        updateState()
+        updateCodeExample()
+    }
+
+    private fun initViews() {
+        initCodeExampleView()
+        setupDateRangesView()
+        binding.mbSubmit.setOnClickListener { submit() }
+        binding.mbGroupCodeExampleType.addOnButtonCheckedListener { _, _, _ -> updateCodeExample() }
+        binding.ccInputsRoot.layoutTransition.enableTransitionType(LayoutTransition.CHANGING)
+        collect.bindView(binding.vgsTiedDateRange)
+    }
+
+    private fun initCodeExampleView() {
+        val syntaxColor = ContextCompat.getColor(this, R.color.veryLightGray)
+        val bgColor = ContextCompat.getColor(this, R.color.blackPearl)
+        val lineNumberColor = ContextCompat.getColor(this, R.color.nobel)
+        codeExampleBinding.cvResponse.setOptions(
+            Options(
+                context = this.applicationContext, theme = ColorThemeData(
+                    SyntaxColors(
+                        string = syntaxColor,
+                        punctuation = syntaxColor,
+                    ),
+                    numColor = lineNumberColor,
+                    bgContent = bgColor,
+                    bgNum = bgColor,
+                    noteColor = syntaxColor
+                )
+            )
+        )
+        codeExampleBinding.cvResponse.alpha = 1f
+        codeExampleBinding.cvResponse.findViewById<RecyclerView>(R.id.rv_code_content).isNestedScrollingEnabled = false
+    }
+
+    private fun setupDateRangesView() {
+        // Specify VGSDateRangeSeparateSerializer to send day, month and year separately in json structure
+        binding.vgsTiedDateRange.setSerializer(
+            VGSDateRangeSeparateSerializer(
+                "date.day",
+                "date.month",
+                "date.year"
+            )
+        )
+        binding.vgsTiedDateRange.setOnFieldStateChangeListener(object : OnFieldStateChangeListener {
+            override fun onStateChange(state: FieldState) {
+                Log.d(CollectActivity::class.java.simpleName, "onStateChange: ${state.fieldName}")
```

Comment:
@DonaldRG Remove `CollectActivity` import and print the whole state, only `fieldName` is not helpful.